atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -431,6 +435,58 @@ public void markTableInvolved(TableReference tableRef) {
         return postFilteredResults;
     }
 
+    @Override
+    public Iterator<Map.Entry<Cell, byte[]>> getSortedColumns(
+            TableReference tableRef, Iterable<byte[]> inputRows, BatchColumnRangeSelection columnRangeSelection) {
+        checkGetPreconditions(tableRef);
+        Iterable<byte[]> rows = Streams.stream(inputRows)
+                .map(ByteBuffer::wrap)
+                .distinct()
+                .map(ByteBuffer::array)
+                .collect(toList());
+        if (Iterables.isEmpty(rows)) {
+            return Collections.emptyIterator();
+        }
+        hasReads = true;
+        int size = Iterables.size(rows);
+        int batchSize = Math.max(
+                Math.min(100, columnRangeSelection.getBatchHint()), columnRangeSelection.getBatchHint() / size);
+        BatchColumnRangeSelection perBatchSelection = BatchColumnRangeSelection.create(
+                columnRangeSelection.getStartCol(), columnRangeSelection.getEndCol(), batchSize);
+        Map<byte[], RowColumnRangeIterator> rawResults =
+                keyValueService.getRowsColumnRange(tableRef, rows, perBatchSelection, getStartTimestamp());
+
+        Iterator<Map.Entry<Cell, Value>> cells = mergeColumnFirst(rawResults.values());
+        Iterator<Map.Entry<Cell, Value>> postFilterIterator =
+                getRowColumnRangePostFiltered(tableRef, cells, columnRangeSelection.getBatchHint());
```

Comment:
As written, this method performs a re-sort on the entries so that they will not be in the column-first order but instead in row query order.

I wrote the following test, which fails:

```
    @Test
    public void getSortedColumns() {
        byte[] row1 = "foo".getBytes();
        byte[] row2 = "bar".getBytes();
        byte[] col1 = "a".getBytes();
        byte[] col2 = "b".getBytes();

        Cell rowOneColumnOneCell = Cell.create(row1, "a".getBytes());
        Cell rowOneColumnTwoCell = Cell.create(row1, "b".getBytes());
        Cell rowTwoColumnOneCell = Cell.create(row2, "a".getBytes());
        Cell rowTwoColumnTwoCell = Cell.create(row2, "b".getBytes());
        byte[] value = new byte[1];

        serializableTxManager.runTaskWithRetry(tx -> {
            tx.put(TABLE, ImmutableMap.of(rowOneColumnOneCell, value, rowOneColumnTwoCell, value,
                    rowTwoColumnOneCell, value, rowTwoColumnTwoCell, value));
            return null;
        });

        List<Cell> entries = serializableTxManager.runTaskWithRetry(tx -> {
            Iterator<Entry<Cell, byte[]>> sortedColumns =
                    tx.getSortedColumns(TABLE, ImmutableList.of(row1, row2), BatchColumnRangeSelection.create(
                            PtBytes.EMPTY_BYTE_ARRAY, PtBytes.EMPTY_BYTE_ARRAY, 1000));
            return Streams.stream(sortedColumns).map(Entry::getKey).collect(Collectors.toList());
        });
        org.assertj.core.api.Assertions.assertThat(entries).containsExactly(
                rowTwoColumnOneCell,
                rowOneColumnOneCell,
                rowTwoColumnTwoCell,
                rowOneColumnTwoCell); // Nope, order is R2C1, R2C2, R1C1, R1C2
    }
```

Basically I think there's some additional processing around this that means we don't want to just grab the same impl from getRowsColumnRangeIterator. I get that a lot of the logic _is_ similar, but there's a fair bit of postprocessing that was written for that method that's kind of specific there.